cvat/apps/engine/serializers.py

```diff
@@ -1196,12 +1196,29 @@ def update(self, instance, validated_data):
         _update_related_storages(instance, validated_data)
 
         instance.save()
+
+        if 'label_set' in validated_data and not instance.project_id:
+            self.update_children_objects_on_labels_update(instance)
+
         instance.task_labels_count = instance.label_set.filter(
             parent__isnull=True).count()
         instance.proj_labels_count = instance.project.label_set.filter(
             parent__isnull=True).count() if instance.project else 0
         return instance
 
+    def update_children_objects_on_labels_update(self, instance: models.Task):
+        # Use light wrappers and load only necessary values to avoid heavy requests
+        updated_jobs = [
+            models.Job(**v)
+            for v in models.Job.objects.filter(
+                segment__task=instance
+            ).values("id", "updated_date")
```

Comment:
Wouldn't `only` be more appropriate here (since you load the values into the model class anyway)?